test_examples.sh

```diff
@@ -20,25 +34,86 @@ function cleanup {
 }
 trap cleanup EXIT
 
+export ALL_OPTS=$TEMPDIR/all_opts
+if [[ "$LONG" == "--long" ]]; then
+  p() { [ $# -eq 0 ] && echo || (shift; p "$@") | while read r ; do echo -e "$1,$r\n$r"; done }
```

Comment:
it generates all possible subsets. Not all permutations. This would be nice, but would take forever to run. This has helped quite a lot in finding bugs.